src/app/search/search.component.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { ActivatedRoute, Router } from '@angular/router';
+import { By, cit, create, cms, provide } from '../../testing';
+import { RouterStub, ActivatedRouteStub } from '../../testing/stub.router';
+import { SearchComponent } from './search.component';
+
+let activatedRoute: ActivatedRouteStub = new ActivatedRouteStub();
+
+describe('SearchComponent', () => {
+
+  create(SearchComponent);
+
+  provide(Router, RouterStub);
+  provide(ActivatedRoute, activatedRoute);
+
+  let auth;
+  beforeEach(() => {
+    activatedRoute.testParams = {};
+    auth = cms.mock('prx:authorization', {});
+    auth.mock('prx:default-account', {});
+    auth.mockItems('prx:series', []);
+    auth.mockItems('prx:stories', []);
+  });
+
+  cit('detects no results', (fix, el, comp) => {
+    expect(el).not.toContainText('Viewing all');
+    expect(comp.noResults).toBe(true);
+  });
```

Comment:
Testing probably needs more work. I found that I couldn't really test what's inside the child components, such as looking for the No stories match your search text. Yet, I'm not sure how much sense it makes sense to test those child components individually because they don't have a lot of responsibility other than displaying what was given to them and emit up events to the parent.